### PR TITLE
Add optional experimentalStreamingLambda field for prerender

### PR DESCRIPTION
**Pull request by nabsul**
This adds a new `experimentalStreamingLambda` field to Prerender outputs, allowing references to an optional streaming lambda path.

### CODE REVIEW

**Comment by cb1kenobi** [comment]
Do we have any tests that exercise this `experimentalStreamingLambda` property?

File: packages/build-utils/src/prerender.ts [hunk 1]
```diff
@@ -130,5 +133,7 @@ export class Prerender {
       }
       this.allowQuery = allowQuery;
     }
+
+    this.experimentalStreamingLambda = experimentalStreamingLambda;
```
Comment (ijjk) [inline packages/build-utils/src/prerender.ts]: We should validate this field the same way we do above for other fields.

**Comment by ijjk** [comment]
@cb1kenobi no tests as this is being validated still and is experimental/might be removed.

File: packages/build-utils/src/prerender.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ interface PrerenderOptions {
   initialStatus?: number;
   passQuery?: boolean;
   sourcePath?: string;
+  experimentalStreamingLambda?: string;
```
Comment (ijjk) [inline packages/build-utils/src/prerender.ts]:
```suggestion
  experimentalStreamingLambdaPath?: string;
```
Maybe we call this the path to stress that's what it is and not a Lambda output itself like the `lambda` field.

File: packages/build-utils/src/prerender.ts (hunk 1 above)
Comment (nabsul) [reply packages/build-utils/src/prerender.ts]: Ah yeah. I forgot that TypeScript doesn't save you from that :-)

File: packages/build-utils/src/prerender.ts (hunk 2 above)
Comment (nabsul) [reply packages/build-utils/src/prerender.ts]: That is a better name!

**Comment by nabsul** [comment]
Thanks for the feedback @cb1kenobi, I went ahead and added a unit test.

**Comment by nabsul** [comment]
Moving to a non-forked PR: https://github.com/vercel/vercel/pull/10476

**Pull request closed without merging**